atlite/cutout.py

```diff
@@ -163,8 +163,15 @@ def __init__(self, path, **cutoutparams):
             logger.info(f"Building new cutout {path}")
 
             if "bounds" in cutoutparams:
-                x1, y1, x2, y2 = cutoutparams.pop("bounds")
-                cutoutparams.update(x=slice(x1, x2), y=slice(y1, y2))
+                bounds = cutoutparams.pop("bounds")
+                # If its a dataframe, we will extract the values
+                if isinstance(bounds, pd.DataFrame):
+                    row = bounds.iloc[0]
+                    x1, y1, x2, y2 = row[["minx", "miny", "maxx", "maxy"]].to_list()
+                else:  # If its a list
+                    x1, y1, x2, y2 = bounds
```

Comment:
Your suggestion would always take the first row, even if the user falsely provides a multi-row dataset.
Instead we should raise an error for the user if the DataFrame has != 1 row.

How about:

```suggestion
                # If its a dataframe, we will extract the values
                if isinstance(bounds, pd.DataFrame) and bounds.shape[0] == 1:
                    x1, y1, x2, y2 = bounds.iloc[0][["minx", "miny", "maxx", "maxy"]].to_list()
                elif isinstance(bounds, tuple):  # If its a list
                    x1, y1, x2, y2 = bounds
                else:
                    raise ValueError("`bounds` must be a tuple or a DataFrame with a single row entry.")
```